tests/adc_mux.test.cpp

```diff
@@ -0,0 +1,93 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "../include/libhal-soft/adc_mux.hpp"
+#include <boost/ut.hpp>
+#include <iostream>
+#include <libhal-mock/adc.hpp>
+#include <libhal-mock/output_pin.hpp>
+#include <queue>
+#include <vector>
+
+namespace hal {
+
+void adc_mux_test()
+{
+  using namespace boost::ut;
+  using level_t = hal::output_pin::level_t;
+  using read_t = hal::adc::read_t;
+  using adc_mux_pin = hal::adc_multiplexer::adc_mux_pin;
+  // configure virtual testing pins
+  const int NUM_PINS = 2;
+  std::array<hal::output_pin*, NUM_PINS> signal_pins = { new mock::output_pin,
+                                                         new mock::output_pin };
+  for (auto& s : signal_pins) {
+    s->level() = level_t{ .state = false };
+  }
+
+  mock::adc* source_adc = new mock::adc();
```

Comment:
```suggestion
  auto source_adc = mock::adc::create();
```

Refrain from using new unless you definitely have to.

Reply:
function doesn't exist. using constructor